cityofaustin/atd-moped
# Add validation of required fields to Team table and combine DataGridTextField components

## Associated issues
Closes https://github.com/cityofaustin/atd-data-tech/issues/20357
Closes https://github.com/cityofaustin/atd-data-tech/issues/20213

This PR updates the Team table to validate required fields, and it combines duplicate `DataGridTextField` components. It also extends the existing validation to use error states that we can get from [preProcessEditCellProps](https://mui.com/x/react-data-grid/editing/?srsltid=AfmBOorAjy4Egs9sAYiz-XwgY_B6xVw3agN4JORVCgeFvxCLA5VKK3q5#validation) in addition to the existing logic that checks for values in the form. So, we can now validate with custom logic on a field-specific level in addition to checking whether there is a value or not. This was needed to handle validating the **Role** field which is an array of values.

## Testing
**URL to test:** 

Local or deploy preview

**Steps to test:**
1. Test the Team table and see if you can get it to save without having a user and at least one role chosen. You should see the check icon greyed otherwise. You will also see red highlighted inputs and helper text on required field once you start editing the form.
2. Test the Milestones table to makes sure the validation on the **Milestone** input still works. Also, test the **Description** input to make sure it still works.
3. Test the Files table to make sure the **Name**, **File** (for a link since this isn't editable for uploaded files), and **Description** inputs still works.
4. Test the Funding table to make sure the **Description** inputs still works.
5. I also made some small tweaks to the styles to keep inputs from butting up against the cell borders and to move some old styles to use the theme spacing. If anything looks off, let me know, and I'll fix or restore if I made things worse.

---
#### Ship list
